ArduCopter/AP_Arming.cpp

```diff
@@ -634,11 +634,9 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
     }
 
 #ifndef ALLOW_ARM_NO_COMPASS
-    // if external source of heading is available, we can skip compass health check
-    if (!ahrs.is_ext_nav_used_for_yaw()) {
-        const Compass &_compass = AP::compass();
-        // check compass health
-        if (!_compass.healthy()) {
+    // if external nav or GPS used for heading, we can skip compass health check
+    if (!ahrs.using_external_yaw()) {
+        if (!AP::compass().healthy()) {
```

Comment:
this gives the wrong error message "Compass not healthy", if using GPS yaw but the GPS doesn't currently have a moving baseline lock

Reply:
the word "using" is confusing, does it mean configured or configured&&healthy

Reply:
I've reduced the scope of this PR so these lines are no longer modified.  I think the issue pointed out is an existing issue and is not made worse by this change.